improvement of residual-chart hover info

ds26gte opened the pull request.

jpolitz committed:
on-raw-key

jpolitz committed:
try reducing this to one variable

jpolitz committed:
trying minimal configurations first

jpolitz committed:
might need a better way to configure these, since we don't know the PR app name up front

jpolitz committed:
try inheriting this from base project

jpolitz committed:
Learned some things about deployment

- app.json is only read when FIRST CREATED, future deploys do
  not read it
- All values must be strings

jpolitz committed:
inherit client id so share URLs work

jpolitz committed:
checking if we can use  here

jpolitz committed:
more testing, this time with explicit branch

jpolitz committed:
Updates to match event design in https://github.com/brownplt/pyret-lang/commit/eadcfdf649d94c11e8c4e6c44c3be80dec9574b4

jpolitz committed:
remove stray console log

Commit:
stylize tildes in roughnums to be more readable.  requires the #emphasize-tildes branch of pyret-codemirror-mode

asolove committed:
Fixes #492 and cleans up unused code.

asolove committed:
Disable checking file name and erroring for shared gdrive files.

jpolitz committed:
Add a new #initialState parameter

Intended for embedding use cases, the type is:

type InitialState = {
    editorContents: string,
    interactionsSinceLastRun: string
}

On load the editor will automatically run definitions with the given
text and then run all the given interactions.

Currently only enabled in embedding contexts.

Need state for two more cases:

- Editing definitions but haven't run at all yet
- Editing definitions after a run

jpolitz committed:
trying to fix travis failures from e.g.https://app.travis-ci.com/github/brownplt/code.pyret.org/builds/271019572

jpolitz committed:
Merge branch 'horizon' into vmt-initialState

jpolitz committed:
support an explicit 'I am ready' event and an initial state receiving event

jpolitz committed:
track message numbers to do state resets in the replayer

jpolitz committed:
initialState and new events

Co-authored-by: Irv Katz <exidy80@users.noreply.github.com>

jpolitz committed:
Some guards and logs around initialization and edits

One thing that's challenging is the use of on('change') on CodeMirror,
which triggers for calls to setValue().

In the events API, we change these to all use replaceRange with an
explicit `origin`. However, there are setValue() calls elsewhere in the
codebase. We could ignore these by checking the origin of "setValue",
but this could be a mistake if we do want to propagate around updates
like those. Need to think about this.

For now, filter out ones that come in before we are initialized, and
wait for the reset event to start sending things out. This makes sure
that the initialization of setting e.g. `use context current` doesn't
get sent out as an event.

Sometimes we get "empty" resets from VMT, and this commit just filters
those out. But it's not clear when exactly that happens.

jpolitz committed:
Some initialization cleanup

- Don't propagate setvalue events; these are coming from page setup,
  but we want to wait for our 'reset'
- Don't setValue at all if the page is “controlled”. Add a new
  #-parameter #controlled=true that tells the page to not do its usual
  initialization and instead wait for the reset events
- The first 'reset' has a state of '', so deal with that appropriately
  to handle initialization in events.js

jpolitz committed:
Pass repl updates around

jpolitz committed:
log the correct values in this check

jpolitz committed:
Committing this built file as part of test deploy to https://pyret-vmt-dfb765867402.herokuapp.com/

jpolitz committed:
Make initial state have newlines again

jpolitz committed:
Add logic to allow initialStates that are #share url

jpolitz committed:
update with clean build

jpolitz committed:
Better event descriptions

jpolitz committed:
rebuild for deploy

jpolitz committed:
some blur() and refresh() to fix little visual bugs

asolove committed:
[blocks] remove user-specified function definition and application

jpolitz committed:
Add config option to proxy shared reqs through another server

This is a total hack, and should probably be disabled. However,
for demo servers it can be incredibly valuable to have share
links from e.g. code.pyret.org work. Currently they won't
because the Google credentials are wrong. But, we can simply
configure these fetches to ask for the corresponding resource on CPO.

Consider trying that, then falling back to looking on this server if needed?

jpolitz committed:
If there's no initial state, *do* click the run button for the user

(definitionsAtLastRun = false means the program has not been run. We
can do one better by setting it to starter2024 and running it)

jpolitz committed:
Use the shared fetch server, but fall back if it fails

jpolitz committed:
unconditionally allow CPO urls

jpolitz committed:
Changes for manual control

- Hide and show run/stop on lose/gain control
- Make editor and REPL non-editable when losing control
- Add event listeners on editor and repl to prompt for control when
  not in control
- Fix a bug where the old REPL contents could be included in the state
  when it should have been empty just after an interaction

jpolitz committed:
Don't start without control

jpolitz committed:
Allow resetFromShareLink to happen while the page is running

jpolitz committed:
Actuall return from promise

jpolitz committed:
Merge branch 'horizon' into vmt-initialState

jpolitz committed:
fix some initialization issues

jpolitz committed:
RPCs (for getting file contents in VScode)

jpolitz committed:
Basic file locators

- Need to make sure we report them as not present if window.MESSAGES isn't
- Lots more we can do with modified time, etc, this is just copying from gdrive
  stuff

jpolitz committed:
Stop, and wait for results from, the last run

The high-level goal here is a clean solution to having an in-control
user move everyone along if a reactor or chart modal is open.

The idea is to trigger a stop if a user is (a) not in control and (b)
a run or runInteraction event comes in. This means that all users can
interact with the chart, reactor, etc on their own, and won't be
interrupted if the in-control user stops theirs. But as soon as the in-
control user moves on to run another interaction or use the Run button,
it will forcibly stop the other clients.

This is *not* perfect. Other designs might work better.

The best case for this design is that people get told in chat that they
should close on their own, and the forced stop is just a way to sync
back up.

Some weaknesses and tradeoffs discussed below:

1. For a main program like this, if the forced stop happens the `r`
   variable ends up not being defined, which isn't great.

    r = reactor:
      init: 1,
      on-tick: _ + 1
    end

    r.interact()

2. We could capture the “close” event of the modals and try to propagate
   it around. This is tricky because:
   - It's hard to account for all the ways the modal can close (click,
     `stop-when` trigger, reactor ending in error, etc)
   - The in-control user's game might end early (because of stop-when),
     and it could be nice to let others continue.

3. We could trigger something less drastic than a full “stop” event. For
   example, we could just do JSworld.shutdown({ clean: true }) when the
   next interaction happens.

   However, a single program can have multiple reactors that open in
   serial, conditional on what actions happened. This happens in
   practice! For example the car simulator in physics starts several
   reactors for several different initial conditions. And in general
   it's nice to have a solution that for sure gives control back to the
   REPL.

4. Another option would be to:

   - Queue the interactions that come in while the reactor is running,
     and run them once the reactor stops.
   - Only do a force stop when the *Run* button is clicked.

   This would make it so folks can independently run their animations,
   and when they quit the next interactions would start to process. If
   the in control user instead re-runs the whole program, *then* all
   others get stopped. The queue is a bit nontrivial to maintain, but
   this may be a better user experience for when a "re-sync" ought to
   happen.

   The queue is also nice architecturally because there are potential
   issues of things taking different times to run on different machines
   for long-running programs. This could be a potential issue for e.g.
   loading sheets lagging on some folks' computers and causing them
   to get stopped unnecessarily, and the “queue until rerun” would help
   a lot with that.

jpolitz committed:
Queue all effects from messages in a single queue, reset on run

This implements strategy #4 from https://github.com/brownplt/code.pyret.org/commit/af6f425221af5774c3b9d6bda46791f9331db191

The idea is that *all* messages (other than gain/lose control) now go
into a queue. Each message's effect is an async thunk (so it returns a
promise) and there's a standard queue-processing loop for it.

The exception is run events (as in clicking the Run button), which have
the special effect of *stopping the program* and *clearing the queue*.
This makes it so clicking “Run” will bring everyone back in sync.

When a modal is open, nothing will force it closed for clients other
than clicking the run button; interactions typed after it by those
in control will queue up and wait for the active user to finish the
modal evaluation themselves (e.g. closing the window, their stop-when
terminates). The promise used to finish a run or runInteraction event
is tied to the termination of the evaluation, so all the ways we have
of stopping evaluation are supported.

A thought – we could potentially re-enable the stop button for users
who aren't in control for long-running or hung programs with this
scheme, because that can be under independent control.

I think that reset() needs to be rewritten in terms of the queue still,
because its loop over the pending interactions will (I think) continue
to run even after the stop() from clicking run. So if a reset event
is followed by a run event, it won't cleanly clear up all the pending
stuff. This is next up.

jpolitz committed:
reset from the current state; don't expect arbitrary messages to have the same state as a reset

jpolitz committed:
bare minimum testing for embedding API (does it send a pyret init

jpolitz committed:
add test for reset event

jpolitz committed:
hide/remove useless elements when in headless mode

jpolitz committed:
Revert "Basic file locators"

This reverts commit d504e15e0ba15c9b39f35073b75b4a8bf6b66594.

This seems to have caused an issue in the CI; not sure what it is but
removing this to see what happens in Travis if we undo it

jpolitz committed:
remove stray logs (that actually log a lot of annoying junk on downloadImg)

jpolitz committed:
Revert "Revert "Basic file locators""

This reverts commit 01b65f3be8303f5b902397cfce4b9eeef9bfd671.

jpolitz committed:
Don't try to use MESSAGES.sendRpc unless it exists

jpolitz committed:
Test with two frames, showing edits propagating across

jpolitz committed:
Properly parameterize tests with the BASE_URL from env

jpolitz committed:
reduce timer length to make failed runs quicker

jpolitz committed:
allow CI to postmessage

jpolitz committed:
try longer timeouts first; surprised these are failing when other embed tests working

jpolitz committed:
update cpo-main

jpolitz committed:
stop before reset

jpolitz committed:
stop before resetting

jpolitz committed:
exit early if no gapi defined, and avoid shadowing fileLocator error

jpolitz committed:
logging, build updates

jpolitz committed:
Make cookies be samesite: strict

This makes it so session cookies won't be enabled when iframed.

This avoids thinking about a whole host of things that can happen for embedded
instances trying to save files, etc. Embedded instances will *always* behave as
if logged out to start, and will not show any UI elements for logging in.

jpolitz committed:
make it so if #initialState is provided, the page is put into control after initializing. *Way* easier API for simple embedding

jpolitz committed:
regenerate package-lock; travis is complaining about npm ci

schanzer committed:
set the title attribute of the filename element to be the FULL filename

schanzer committed:
tweak CSS for even-numbered rows to make them more apparent (#185)

jpolitz committed:
does this work with just plain install?

jpolitz committed:
respond to several review comments

jpolitz committed:
built files (we won't always need to include these, just for intermediate deploys) after testing last commit

jpolitz committed:
does this succeed if we just update pyret-lang and don't update all packages?

jpolitz committed:
npm update

jpolitz committed:
bundle events in an abstraction

jpolitz committed:
only update pyret-lang

jpolitz committed:
only update pyret-lang

jpolitz committed:
only update pyret-lang

jpolitz committed:
back to lax for testing

jpolitz committed:
npm update

jpolitz committed:
npm i after update

jpolitz committed:
now strict again for more live testing

jpolitz committed:
back to lax for testing

jpolitz committed:
Merge branch 'horizon' into vmt-initialState

jpolitz committed:
remove this to test deploy

jpolitz committed:
do git updates first before building to fetch snap

jpolitz committed:
remove submodule attempt

jpolitz committed:
Revert "remove this to test deploy"

This reverts commit 0b0038092b524e764559507c240e0395ffce9fc7.

jpolitz committed:
Merge pull request #569 from brownplt/vmt-initialState

VMT/embedding API

jpolitz committed:
Merge pull request #572 from brownplt/table-contrast

tweak CSS for even-numbered rows to make them more apparent (#185)

schanzer committed:
do the same for beforePyret.js

jpolitz committed:
Merge pull request #571 from brownplt/file-title

set the title attribute of the filename element to be the FULL filename

jpolitz committed:
remove built file because horizon can deploy itself

jpolitz committed:
Track controlled-ness and use it to manage focus

- Add explicit state that tracks if we are in control or not, and don't disable
  editors if we are in control
- After setting interactions, if we're in control, focus it.

jpolitz committed:
if definitionsAtLastRun is false on reset, run the empty program, which is the right re-simulation of starting from scratch

jpolitz committed:
Some conservative embedders (like github.dev) require this

I think this is the issue, I'm getting this when using the published VScode web
extension and embedding CPO.

"not-set cross-origin-resource-policy:

Because your site has the Cross-Origin Embedder Policy (COEP) enabled, each
resource must specify a suitable Cross-Origin Resource Policy (CORP). This
behavior prevents a document from loading cross-origin resources which don’t
explicitly grant permission to be loaded."

The COEP is enabled by github.dev (I think), so this says “yup please embed us”

jpolitz committed:
More diagnosing issues from embedding in github.dev

"not-set cross-origin-embedder-policy:

To embed this frame in your document, the response needs to enable the cross-origin embedder policy by specifying the following response header:

Cross-Origin-Embedder-Policy: require corp
"

jpolitz committed:
Start of CSV parsing with fast-csv (https://www.npmjs.com/package/fast-csv)

jpolitz committed:
csv-parser branch for this branch

jpolitz committed:
Sketching out file-image

jpolitz committed:
Starting to fill in the fs library with rpcs

If we have a sendRpc target (like VScode) we can try to fill in the fs library
to some extent.

We won't be able to get full coverage because we can't do synchronous RPCs (so
e.g. filelib won't work transparently)

But we can do a lot of it, and we can probably update the pyret-lang files to
use a reasonable subset that we can polyfill with RPCs to embedding contexts
that do have a filesystem

jpolitz committed:
`reportAnswer` on runInteraction and more inControl

- Add a `reportAnswer` field to `runInteraction` that requests the interaction
  report its result back. For now, this is the innerText of the last element of
  the #output, which is probably enough for a lot of things, but a little kludgy.
- For the “recovery”code that runs when state gets desynced, ONLY run it when we
  are not in control. For embedded instances that are allowing editing, it's
  better to just trust our current state rather than expecting embedders to
  keep all state and message sequencing up to date.

jpolitz committed:
Basic image-file(<path>) support

- Relies on readFile from the embedding context that gives back the Uint8Array,
  rather than the UTF8-stringified result of the file
- Adds image-file, which gets this Uint8 buffer from a path and converts it to
  base64
- Uses a (small for now) hardcoded list of extensions to trust. This is what
  browsers do in a file upload dialog (see
  https://developer.mozilla.org/en-US/docs/Web/API/Blob/type, “Note: Based on
  the current implementation, browsers won't actually read the bytestream of a
  file to determine its media type. It is assumed based on the file extension;
  a PNG image file renamed to .txt would give "text/plain" and not "image/png".
  Moreover, blob.type is generally reliable only for common file types like
  images, HTML documents, audio and video.”)

jpolitz committed:
Update to match node's readFile spec

Pass an empty opts to indicate that raw Uint8 data should come back, rather
than having a separate endpoint.

jpolitz committed:
undo stray edit

jpolitz committed:
better error messageg

jpolitz committed:
Merge pull request #576 from brownplt/csv-parser

csv-parser library updates

jpolitz committed:
Merge pull request #577 from brownplt/image-file

image-file

ds26gte committed:
Improved hover info on dots on residual charts

ds26gte committed:
interval chart: Significant digits on delta shd match y, ŷ

ds26gte committed:
interval chart: Restrict signif digits to 4

ds26gte committed:
Merge pull request #578 from ds26gte/residual-hover-info

Residual hover info

jpolitz committed:
pyret version, match pyret version

jpolitz committed:
package-lock

jpolitz committed:
actually pass arguments to trigger handlers

jpolitz committed:
.

jpolitz committed:
Move snap to a npm package

jpolitz committed:
Merge pull request #547 from asolove/blocks-mvp

[Blocks] remove user-specified function definition and application

jpolitz committed:
Merge pull request #494 from asolove/asolove/publish-dialog

[WIP] Share modal updates

jpolitz committed:
update tests for pyret-lang wording changes around predicate tests

jpolitz committed:
update tests for pyret-lang wording changes around predicate tests

jpolitz committed:
Merge pull request #581 from brownplt/snap-npm

Move snap to a npm package

jpolitz commented:
The title of this PR is (now?) broader than what is contained, since this is a full horizon -> mainmast update.

Is there a specific commit to cherry-pick that someone really needs? Maybe there's no reason to not deploy everything? Some extra commentary would help

jpolitz committed:
kick to deploy

jpolitz committed:
since this is in an npx-ish context, the webpack executable *should* be fine

jpolitz committed:
Progress on an all-local setup for built CPO

Design goals:

- Make it so CPO runs with the network on your device *turned off*
- Clearly identify all the dependencies we need for a running CPO instance
- More easily support (primarily for a VScode extension-like context) an
  all-local built bundle

This initial commit does a few things:

1. Downloads relevant 3rd party JavaScript (fonts, and JQuery stuff) and copies
it in Makefile
2. Does some housekeeping on different failure cases to make the page load even
if e.g. Google scripts don't load

After building, you can try this out for now by setting `PYRET` to e.g.
`/js/cpo-main.jarr`, then

node make-template src/web/editor.hml > build/web/editor.html
cd build/web/
python3 -m http.server 8080
// turn off your network and open localhost:8080/editor.html

Note that right now the *.html files are *dynamically* templated, so the extra
build step (which maybe should unconditionally happen) creates the
correctly-filled-in editor.html

Some things simply cannot be made to work in this way. The chart loader and
google API loader I left alone; even if you download those files and load them
from source locally they immediately make network requests, embed a 3rd party
iframe, etc. So it's unlikely that stuff will ever work offline.

It's an open (and pretty important) question if the Google Chart library has a
good offline build mode so that e.g. you can connect CSV-file charts to that
library

jpolitz committed:
More protections around google charts not loading

jpolitz committed:
lost some config in recent updates

jpolitz committed:
better templating for the question mark glyph

jpolitz committed:
this somehow did _not_ get updated with the readFileSync->readFile update in the extension API

jpolitz committed:
this somehow did _not_ get updated with the readFileSync->readFile update in the extension API

jpolitz committed:
attempting to run everything just depending on pyret-lang's image definition

jpolitz committed:
update images tests

jpolitz committed:
Make canvas.Image be just the native Image in browser

jpolitz committed:
Merge branch 'local-deps' into offline-image

jpolitz committed:
writefile

jpolitz committed:
Allow templating in hash params

Some contexts, like Webviews, don't have a traditional document.location, or
don't allow it to be easily set.

In those contexts, allow templating in the hash params

jpolitz committed:
Add hideInteractions as well as hideDefinitions

hideInteractions is special: we don't have a good way to show run results
(other than highlighting and gutter marks).

So, as soon as run is clicked, it shows interactions again. But this is useful
for having e.g. many editors open in different extension tabs, or just
embedding a plain editor.

jpolitz committed:
Utility script for getting images in other scripts

jpolitz commented:
@dorai I think you're the person who knows about my last comment?

jpolitz commented:
@ds26gte I think you would know best about my last comment?

jpolitz committed:
Merge pull request #518 from asolove/asolove/name-mismatch

Disable checking file name and erroring for shared gdrive files.

jpolitz committed:
test for loading a starter file

jpolitz committed:
more chromedriver updates

jpolitz committed:
stray log

jpolitz committed:
Increasing embed test timeouts: sometimes takes a while to load

jpolitz committed:
Increase timeout and allow looking up files on CPO

jpolitz committed:
Add publicOnly as a flag when initializing Drive storage

This is the start of some refactoring and cleanup around page loads.

When a main CPO window starts up, it unconditionally makes a few requests to
figure out the login state and where the user's programs should be saved,
cached, etc within Drive.

In embedded or VScode contexts, this makes no sense at all: the cookie policy,
API settings are completely turned off any private access to drive. (Like, at a
browser level we leave on the settings that prevent the iframe from setting
cookies)

As a result, there can be really confusing error messages in the console where
stuff just doesn't load and there are a bunch of apparent 404's. Not great when
you're polluting the console of the embedded page for debugging reasons,
either.

So this is the start of some work to more programmatically “turn off” the
stuff that wouldn't work by design anyway to have things look a little
cleaner/do less useless work.

jpolitz committed:
Merge branch 'horizon' into local-deps

jpolitz committed:
Merge pull request #585 from brownplt/local-deps

Progress on an all-local setup for built CPO

jpolitz committed:
offline-image -> horizon for package.json, package-lock update

jpolitz committed:
url(...) dependency locators for CPO

Conveniently, cross-fetch is one of the better polyfills, so the work on
pyret-lang pretty transparently pays off here. We can just import the pyret
implementation of url-locator and call it, and the underlying libraries do the
right kind of browser fetch()

jpolitz committed:
basic url-file that always concatenates and does a remote fetch

jpolitz committed:
refer to url-import branch of pyret-lang

jpolitz committed:
Basic url-file implementation for CPO

This expects the `fs.exists` message to be present on the embedder to check for
file presence.

The option is templated in for now, which the extension has control over when
it first renders the page. Pages may need to be reloaded for the settings
change to take effect.

I noticed/remembered while doing this that the CPO file import does not track
working directories, this should be fixed by adding to the module finder
context just like in cli-module-loader.arr in pyret-lang.

jpolitz committed:
URI calculation function used in errors/editors, and be sure this is set on window

jpolitz committed:
stray debugger

jpolitz committed:
Add #shareurl= to initial params

This allows loading from a URL like

https://raw.githubusercontent.com/jpolitz/sample-pyret-project/refs/heads/main/lib.arr

as starter code.

This works even in embedded contexts where the usual storageAPI does not load.
It is treated as a “normal” shared file for logged in users, who can save the
program (to Drive only) as another name.

jpolitz committed:
test url imports

jpolitz committed:
point to correct branch

jpolitz committed:
Add source of url and url-file imports to documents directly from moduleFinder

jpolitz committed:
Update pyret-lang to horizon now that url-import merged

jpolitz committed:
Merge pull request #587 from brownplt/url-import

url(...) dependency locators for CPO

jpolitz committed:
crossorigin=anonymous on PYRET loads

VScode-in-browser has strict content security policy rules.

One of them requires that we serve /editor with certain cross-origin rules (see
server.js)

I didn't fully tie the knot on making sure that if PYRET is cross-origin (which
it is for pyret-horizon!) the requests from preload and dynamic loads check the
right header boxes.

jpolitz committed:
missed lookup (params are in the 'get' subdict)

jpolitz committed:
Also copy to cpo-main.jarr.js because browsers like the extension to match the MIME (or infer it from extension)

jpolitz committed:
clean up this logic

jpolitz committed:
kick to deploy

jpolitz committed:
trying edge: true to update to the new dpl2 in Travis. Getting deploy install errors and they had an update

4.00sInstall deployment dependencies
Installing deployment dependencies
Installing gem dependencies: nokogiri (~> 1.15), aws-sdk-s3 (~> 1), mime-types (~> 3.4.1)
GemWrappers: Can not wrap not executable file: /home/travis/.rvm/gems/ruby-3.3.5/bin/racc.lock
GemWrappers: Can not wrap not executable file: /home/travis/.rvm/gems/ruby-3.3.5/bin/racc.lock
GemWrappers: Can not wrap not executable file: /home/travis/.rvm/gems/ruby-3.3.5/bin/racc.lock
GemWrappers: Can not wrap not executable file: /home/travis/.rvm/gems/ruby-3.3.5/bin/racc.lock
You have already activated logger 1.6.0, but your Gemfile requires logger 1.6.6. Since logger is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports logger as a default gem. (Gem::LoadError)
failed to deploy

https://app.travis-ci.com/github/brownplt/code.pyret.org/builds/274663286#L1246

https://docs.travis-ci.com/user/deployment-v2/providers/heroku/

https://www.travis-ci.com/blog/2019-08-27-deployment-tooling-dpl-v2-preview-release/

jpolitz committed:
New template option: IMAGE_PROXY_BYPASS

By default CPO tries to proxy images through /downloadImg?<url>

This allows skipping that entirely (of course, proxying was happening because
many servers don't have CORS enabled)

jpolitz committed:
Missing image files from moving dependencies offline

jpolitz committed:
With checker updates, need to ensure that checks: "main" is used so gdrive/url-file imports don't report test results

jpolitz committed:
try dpl2 for both?

jpolitz committed:
logge(r)

jpolitz committed:
clearInteractions event

jpolitz committed:
enhance reportAnswer results with both text and HTML versions

jpolitz committed:
kick to deploy

jpolitz committed:
Merge pull request #589 from brownplt/emphasize-tildes

Stylize tildes in roughnums to be more readable

Commit:
forgot a file, to improve rendering in the REPL

jpolitz committed:
Fixed relative paths and css/image library build

jpolitz committed:
move image files into lib/css/images/

jpolitz committed:
css/images directory wasn't being created

jpolitz committed:
during build, make editor.embed.html, which is templated with relative paths and just basics, suitable for opening as a static page

jpolitz committed:
Rendering for truncated tables (a new valueskeleton constructor)

jpolitz committed:
these need to go at the end after making build/

jpolitz committed:
kick to deploy

jpolitz committed:
don't check for current version when embedded

jpolitz committed:
kick to deploy

jpolitz committed:
kick to deploy

jpolitz committed:
Refactoring Makefile to work when installed as a nested node module

jpolitz committed:
More node_modules path refactoring

jpolitz committed:
udpate pyret-cm-mode

jpolitz committed:
npx instead of npm exec

jpolitz committed:
install uglify as a dependency explicitly (it wasn't before?)

jpolitz committed:
Merge branch 'horizon' into on-raw-key

jpolitz committed:
make sure we don't overwrite travis with old per-branch build experiments

jpolitz committed:
update pyret-lang

jpolitz committed:
stack safety for uriFromDependency

jpolitz committed:
refactor RPC callbacks into a shared helper

jpolitz committed:
correctly forward exceptions on from rpc responses

jpolitz committed:
Quality of life improvement for JS errors that leak

This would give the “unexpected exception check console” message, but it
wouldn't always get to the console because of errors in the rendering of the JS
error!

This makes sure the JS error (a) shows up in the error message, which is
ultimately more helpful and (b) is logged in full detail with its stack trace
attached.

jpolitz committed:
Starting to move CPO over to use new filesystem infrastructure

jpolitz committed:
add buffer and add more fs support, jsfile support

jpolitz committed:
exists fixup

jpolitz committed:
join

jpolitz committed:
More explicit RPC checking, move an exists check over

jpolitz committed:
better detection of event type (don't rely on data field presence), and more fs-internal

jpolitz committed:
unsaved bad merge

jpolitz committed:
fix search to be 'persistent' so it doesn't close on search

jpolitz committed:
Add a param for #theme

This allows embedders to control the theme. The hash param theme overrides the
user setting, but does not change the stored user setting.

The related change in CPO controls this – instead of eagerly changing the user
setting to the set param value, instead let them differ until the user
explicitly changes it with a UI action.

schanzer committed:
fix pie chart handling where the legend paginates

jpolitz committed:
vega-charts infrastructure for CPO

dbp committed:
Port in .env.example is now 4999; fix README.md

jpolitz committed:
Create makefile.yml

jpolitz committed:
trying updated workflowfile (thanks cline)

jpolitz committed:
pyret should be on horizon

jpolitz committed:
updated with default env vars

jpolitz committed:
targeting some specific tests that are failing

jpolitz committed:
use npm run rather than npx

jpolitz committed:
trying to not use xvfb because headlesss chrome should Just Work tm (or at least checking if that's the case)

jpolitz committed:
Overall timeout and try a known-working test

jpolitz committed:
spelling

jpolitz committed:
remove some tests that depend on google apis for now (no credentials) and try running all tests

jpolitz committed:
set postmessage origin to permissive

jpolitz committed:
Merge pull request #594 from dbp/readme-port-fix

Port in .env.example is now 4999; fix README.md

jpolitz committed:
try to connect to AWS S3

jpolitz committed:
only assume role if doing webby OIDC thing. access key doesn't use that. move up AWS stuff for faster test

jpolitz committed:
test copying built files to AWS

jpolitz committed:
trying sync; --recursive only a thing on cp

jpolitz committed:
special sync rule to make sure browsers see this as a JS file that is gzipped

jpolitz committed:
Trying out automated deploy on the Heroku side (empty commit)

jpolitz committed:
Update makefile.yml

Trying to only run AWS stuff if PR (not sure what happens if it's a push to an existing PR, see https://stackoverflow.com/questions/66205887/only-run-github-actions-step-if-not-a-pull-request)

jpolitz committed:
Update makefile.yml

One more conditional-on-PR-step

jpolitz committed:
these names were not added to the CPO context

jpolitz committed:
Create deploy-trigger.yml

jpolitz committed:
Update makefile.yml

jpolitz committed:
other configuration made these snap files no longer be present

jpolitz committed:
Update makefile.yml

jpolitz committed:
I guess this file has to exist?

blerner commented:
I'm closing this PR as obsolete (and also much too big and conflicting to merge cleanly anymore).  There appear to be three commits that were relevant:
- https://github.com/brownplt/code.pyret.org/pull/579/commits/ef460e37c4ff3fbe90620c7797b68f06e00feddc, which is ported to the vega library already,
- https://github.com/brownplt/code.pyret.org/pull/579/commits/a8e5ace9235242818f6dc389d91f2b4d78508c64 and https://github.com/brownplt/code.pyret.org/pull/579/commits/72ed3e336eaa089f01a54e08a540c7b3a1d5b00d, which have not...but which may or may not be needed, depending on what @schanzer thinks of the new UI
